storage_controller/src/scheduler.rs

```diff
@@ -236,6 +344,7 @@ impl Default for ScheduleMode {
 
 // For carrying state between multiple calls to [`TenantShard::schedule`], e.g. when calling
 // it for many shards in the same tenant.
+// TODO: it would be nice if we could make this shard specific at the type level somehow
```

Comment:
not sure I grok this comment -- shard-specific how?

Reply:
I thought I removed this. It's unclear. It's about enforcing that `SchedulerContext` is not incorrectly reused across tenants.

Reply:
https://github.com/neondatabase/neon/pull/9083/commits/416ad12b80ba488e3de524c4b801d12dfb14f9de